### PR TITLE
Added sides picking and folder processing

### PR DESCRIPTION
**Pull request by thegalkin**
Hi, added two features sides picking and folder processing.
I also had to add a uilive lib to update terminal output dynamically. 
Could you please check them and add if you find them a useful contribution. 

P.S. Sorry these are not the optimizations you included in readme but I needed them for personal usage.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -2,56 +2,190 @@ package cmd
...
+			if inFilePath != "" {
+				progress.totalFiles = 1
+				processSingleImage(inFilePath, outFileDir)
+			}
 
-			fmt.Println("Write images...")
-			if err := conv.WriteImage(canvases, outFileDir, ext); err != nil {
-				er(err)
+			if inDirPath != "" {
+				processDirectory(inDirPath, outFileDir)
```
Comment (blackironj) [inline cmd/root.go]: you wrote the following in the flag desc in the Readme.
```
  -i, --in string    input image file path (required if --indir is not specified)
  -d, --indir string input directory path (required if --in is not specified)
```
If this is not what you intended, how about modifying codes like this?
```go
if inFilePath != "" {
     progress.totalFiles = 1
     progressSingleImage(inFilePath, outFileDir)
}  else {
    processDirectory(inDirPath, outFileDir)
}
```

**Review by blackironj — CHANGES_REQUESTED** [review]
Thank you for contributions!
I think it's a good feature.

please check a single comment.

File: cmd/root.go (hunk 1 above)
Comment (thegalkin) [reply cmd/root.go]: Thank you for your comment! Futhermore, I now see that there is no protection for case when both inFilePath and inDirPath are filled.
```
if inFilePath == "" && inDirPath == "" {
				er("Need an input image file path or input directory")
			}
if len(inFilePath) > 0 && len(inDirPath) > 0 {
er("Need only one path, not both")
}

			progress.startTime = time.Now()

			if inFilePath != "" {
				progress.totalFiles = 1
				processSingleImage(inFilePath, outFileDir)
			} else {
				processDirectory(inDirPath, outFileDir)
			}
```

Comment (blackironj) [reply cmd/root.go]:
```go
if len(inFilePath) > 0 && len(inDirPath) > 0 {
   er("Need only one path, not both")
}
```
I think that  codes what you added above are better. could you add that?

**Review by blackironj — APPROVED** [review]
Thank you

**Comment by thegalkin** [comment]
Thank you too!

**Pull request merged**